src/evaluation/Classes/CharHitAtK.py

```diff
@@ -1,50 +1,62 @@
 import math
 from typing import List
 
+from tqdm import tqdm
+
 from src.classes.model_result import ModelResult
 from src.classes.text_part import TextPart
 from src.evaluation.Classes.HitAtK import HitAtK
+from src.model.ensemble_v2 import EnsembleV2
 from src.model.model import Model
 
 
 class CharHitAtK(HitAtK):
 
-    def calculate(self, model: Model, data: str or List[dict], k: int) -> int:
+    def calculate(self, model: Model, data: str or List[dict], k: int,char_weight:float=None) -> float:
         """
+        ** this hit@k works only for same word length models **
         calculate hit@k score for chars.
         :param model: model to be tested
         :param data: data to be tested on. could be string if its a path to test json file or dict if its a ready to go
                      list in form of [{"text": "...", "missing": {...}}]
         :param k: the k of hit@k
         :return: hit@k score (# of char hists at k)/(# of missing chars)
         """
-        if type(data) == 'str':
+        if isinstance(data, str):
             data = self.get_data_at_hit_at_k_test_format(data)
-
+        results=[]
+        progress_bar = tqdm(range(len(data)), desc=f"{model.model_path} Char Hit@{k}", unit="entry",
```

Comment:
this is cool 👍 
![image](https://user-images.githubusercontent.com/63791132/236632296-00370e49-dc7f-4b5d-bd95-95fc4f851cde.png)